### PR TITLE
CFE-462: Using mock data for testing front page

### PR DESCRIPTION
**Pull request by manikandan-ravikumar**
Currently the nosto-react tests using production account for testing recommendation rendering. This PR updates the tests to use the local mock data instead

Following tests are yet to be migrated to using mock data. These migration will be handled in a separate PR
1. useLoadClientScript
2. nosto.load (to be renamed as well)
3. scriptLoader

### CODE REVIEW

File: spec/home.spec.tsx [hunk 1]
```diff
@@ -1,24 +1,53 @@
-import { test, expect } from "vitest"
+import { test, expect, vi } from "vitest"
 import { render, screen } from "@testing-library/react"
 import { NostoProvider, NostoHome, NostoPlacement } from "../src/index"
 import RecommendationComponent from "./renderer"
 import { waitForRecommendations } from "./utils"
+import { jsonMockDataForPage } from "./mocks"
 
 test("Home page render", async () => {
+  const actionMock = {
```
Comment (timowestnosto) [inline spec/home.spec.tsx:9]: can you move lines 9-36 also to mocks?
and already sketch how this mock is parametrized

File: spec/home.spec.tsx [hunk 2]
```diff
@@ -1,24 +1,53 @@
...
+      })
+    })
+  }
+
+  // @ts-expect-error type mismatch of partial
+  window.nostojs = cb => cb(mockApi)
+
   render(
-    <NostoProvider account="shopify-11368366139" recommendationComponent={<RecommendationComponent />}>
+    <NostoProvider account="shopify-00000" recommendationComponent={<RecommendationComponent />} loadScript={false}>
```
Comment (timowestnosto) [inline spec/home.spec.tsx:39]: shopify-00000 -> dummy
that way it's clearer that it's dummy data

File: spec/home.spec.tsx [hunk 3]
```diff
@@ -1,24 +1,53 @@
...
+    viewFrontPage: () => actionMock
+  }
+
+  const mockApi = {
+    placements: {
+      injectCampaigns: vi.fn(),
+      getPlacements: () => ["frontpage-nosto-1", "frontpage-nosto-2"]
+    },
+    defaultSession: () => ({
+      setVariation: (variation: string) => sessionMock,
```
Comment (timowestnosto) [inline spec/home.spec.tsx:26]: drop parameter declaration

File: spec/home.spec.tsx [hunk 4]
```diff
@@ -1,24 +1,53 @@
...
+      setResponseMode: (mode: string) => sessionMock,
```
Comment (timowestnosto) [inline spec/home.spec.tsx:27]: drop parameter declaration

File: spec/home.spec.tsx [hunk 5]
```diff
@@ -1,24 +1,53 @@
...
+      viewFrontPage: () => ({
+        setPlacements: (placements: string[] = []) => actionMock,
```
Comment (timowestnosto) [inline spec/home.spec.tsx:29]: drop parameter declaration

File: spec/home.spec.tsx (hunk 1 above)
Comment (manikandan-ravikumar) [reply spec/home.spec.tsx:9]: Yes, it's already planned for when I spread the usage. Thanks!

File: spec/home.spec.tsx [hunk 6]
```diff
@@ -1,24 +1,53 @@
...
+    setPlacements: vi.fn()
+  }
+
+  const sessionMock = {
+    setVariation: this,
+    setResponseMode: vi.fn(),
+    viewFrontPage: () => actionMock
+  }
+
+  const mockApi = {
```
Comment (timowestnosto) [inline spec/home.spec.tsx:20]: looks like the parameters for this mock are
* placements: ["frontpage-nosto-1", "frontpage-nosto-2"]
* action results: jsonMockDataForPage("front")

File: spec/mocks/mock-api.ts [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+        }
+      }
+    }
+    default: {
+    }
+  }
+}
+
+export default function (pageType: string, placements: string[]) {
+  const data: Data = {
```
Comment (manikandan-ravikumar) [inline spec/mocks/mock-api.ts:148]: Used for validating the data input from the session API.

File: spec/mocks/mock-api.ts [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+  cart?: unknown
+  customer?: Customer
+  searchTerms?: string[]
+}
+
+function normaliseProduct(data: ProductTagging | string) {
+  return typeof data === "string" ? { product_id: data } : data
+}
+
+function getPageAction(pageType: string, placements: string[], state: Data) {
```
Comment (timowestnosto) [inline spec/mocks/mock-api.ts:32]: can you try to extract a newAction function?
there is too much structural repetition in this function

File: spec/mocks/mock-api.ts [hunk 9]
```diff
@@ -0,0 +1,128 @@
+import { vi } from "vitest"
+import { jsonMockData } from "./mock-data"
+import { Product } from "../../src"
+import { Action, ActionResponse, Cart, Customer, Data, RenderMode } from "../../src/types"
+
+type LoadAction = {
+  load: () => Promise<Partial<ActionResponse>>
+}
+
+type PlacementAction = {
```
Comment (manikandan-ravikumar) [inline spec/mocks/mock-api.ts:10]: Tried to use the `Action` type but it gets a little complex as nosto-react library doesn't use all the apis available in the Action type

**Review by timowestnosto — APPROVED** [review]

**Review by TomJas-nosto — APPROVED** [review]

**Pull request merged**